gulpfile.js

```diff
@@ -27,6 +19,84 @@ const plCompile = require('./tools/tasks/pl-compile')(plPath);
 gulp.task('compile:pl', plCompile);
 
 /**
+ * Gulp namespace, ensures that ./tools/pattern-lab/config.yml & ./theme.info.yml
+ * are updated with all pattern namespaces for error-free compiling.
+ */
+const twigNamespaces = require('./tools/tasks/gulp-twig-namespaces');
+
+gulp.task('compile:twig-namespaces', () => gulp
+  .src('./source/_patterns/**/*.twig')
+  .pipe(twigNamespaces({
+    // Namespaces and required information
+    sets: {
+      base: {
+        root: 'source/_patterns/00-base',
+        ignore: '/demo',
+      },
+      atoms: {
+        root: 'source/_patterns/01-atoms',
+        ignore: '/demo',
+      },
+      molecules: {
+        root: 'source/_patterns/02-molecules',
+        ignore: '/demo',
+      },
+      organisms: {
+        root: 'source/_patterns/03-organisms',
+        ignore: '/demo',
+      },
+      templates: {
+        root: 'source/_patterns/04-templates',
+        ignore: '/demo',
+      },
+      pages: {
+        root: 'source/_patterns/05-pages',
+        ignore: '/demo',
+      },
+    },
+    // Which files to read and overwrite with namespace info
```

Comment:
I would switch the order of written files here, or at least include a note that the `'./tools/pattern-lab/config/config.yml'` object is **mandatory** for pattern lab to compile correctly automatically. The .info file only needs to be written for inclusion of patterns in drupal templates

Reply:
@tjheffner, just pushed new notes here, let me know what you think.